src/codegen/mod.rs

```diff
@@ -1264,11 +1268,21 @@ impl CodeGenerator for Enum {
                     if enum_ty.name().is_none() {
                         // NB: if we want to do this for other kind of nested
                         // enums we can probably mangle the name.
-                        if item.is_toplevel(ctx) {
-                            add_constant(enum_ty, &name, &variant_name,
-                                         &variant_name, enum_rust_ty.clone(),
-                                         result);
-                        }
+                        let mangled_name = if item.is_toplevel(ctx) {
+                            variant_name.clone()
+                        } else {
+                            if parent_canonical_name.is_none() {
+                                parent_canonical_name = Some(item.parent_id().canonical_name(ctx));
```

Comment:
Can we eagerly initialize this to `Some` above the loop if `!item.is_toplevel(ctx)`, rather than unconditionally initializing to `None` and repeatedly checking on every iteration?